src/sentry/api/authentication.py

```diff
@@ -60,11 +72,24 @@ def authenticate_credentials(self, relay_id, relay_sig, request):
         with configure_scope() as scope:
             scope.set_tag("relay_id", relay_id)
 
-        try:
-            relay = Relay.objects.get(relay_id=relay_id)
-            relay.is_internal = is_internal_relay(request, relay.public_key)
-        except Relay.DoesNotExist:
-            raise AuthenticationFailed("Unknown relay")
+        # first see if we have a statically configured relay and therefore we don't
+        # need to go to the database for it
+        relay_options = options.get("relay")
+        relay_info = safe.get_path(relay_options, "static_auth", relay_id)
+
+        if relay_info is not None:
+            # we have a statically configured Relay
+            relay = Relay(
+                relay_id=relay_id,
+                public_key=relay_info.get("public_key"),
+                is_internal=relay_info.get("internal") is True,
+            )
+        else:
+            try:
+                relay = Relay.objects.get(relay_id=relay_id)
+                relay.is_internal = is_internal_relay(request, relay.public_key)
```

Comment:
We repeat ourselves here, by first checking the options, and then doing it again in `is_internal_relay`. There are only two places where `is_internal_relay` is being used, so it would be great to change or split up the function to avoid the repetition.

Reply:
True, is_internal_relay it is really only used for non static relays ( in registration and authentication).

Checking for static internal relays is pointless, a better approach is to change the name of the function to make it clear that it only checks for non static relays and remove the part where it validates static relays.